src/XMLHttpRequest/XMLHttpRequest/createXMLHttpRequestOverride.ts

```diff
@@ -325,7 +324,9 @@ export const createXMLHttpRequestOverride = (
             originalRequest.onerror = this.onerror
             originalRequest.ontimeout = this.ontimeout
             originalRequest.onreadystatechange = this.onreadystatechange
-
+            if (this.async) {
+              originalRequest.timeout = this.timeout
```

Comment:
Should we always inherit `.timeout` callback? I feel it's the end user's responsibility to align that they assign `timeout` callback only for `async` XHR.

Reply:
If we assign a timeout function when a request is not async an error is thrown. We can leave this error for the user

Reply:
We can assign timeout function only if is passed. In this way the error will be thrown if xhr is not used in the right way